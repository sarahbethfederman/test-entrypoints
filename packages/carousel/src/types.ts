import { LUIGlobalProps } from '@lendi-ui/utils';
import { CSSProperties, ReactElement, FunctionComponent } from 'react';

export interface CarouselSession {
  isNextDisabled: boolean;
  isPreviousDisabled: boolean;
  nextSlide: () => void;
  previousSlide: () => void;
  setSlide: (newSlide: number) => void;
  slides: number;
  currentIndex: number;
  increment: number;
  infinite: boolean;
  containerBox: ClientRect | DOMRect;
}

export interface SlidesProps extends LUIGlobalProps {
  setSlides: (slides: number) => void;
  height: string;
  speed: number;
  windowWidth: number;
}

export interface SlidesState {
  width: number;
  currentIndex: number;
  previousIndex: number;
  slideWidths: number[];
}

export interface IncrementMap {
  mobile: number;
  tablet: number;
  desktop: number;
}

export interface CarouselProps extends LUIGlobalProps {
  width?: string;
  height?: string;
  initialIndex?: number;
  increment?: number | IncrementMap;
  withArrows?: boolean;
  withDots?: boolean;
  infinite?: boolean;
  autoplay?: boolean;
  duration?: number;
  speed?: number;
  swipe?: boolean;
}

export interface CarouselArrowsProps extends LUIGlobalProps {
  width?: number;
  children?: ((disabled: boolean) => ReactElement) | ReactElement;
}

export interface CarouselDotsProps extends LUIGlobalProps {
  children?: ((isActive: boolean) => ReactElement) | ReactElement;
  style?: CSSProperties;
}

export interface CarouselArrowOffset {
  x: number;
  y: number;
}

export interface CarouselArrowStyleProps {
  disabled?: boolean;
  offset?: CarouselArrowOffset;
  hide?: boolean;
}

export interface CarouselDotsStyleProps {
  isActive?: boolean;
}

export interface CarouselCompoundComponent {
  Slides: FunctionComponent;
  NextArrow: FunctionComponent;
  PreviousArrow: FunctionComponent;
  Dots: FunctionComponent<CarouselDotsProps>;
}
