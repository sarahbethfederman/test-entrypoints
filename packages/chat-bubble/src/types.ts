import { LUIGlobalProps } from '@lendi-ui/utils';

export type Direction = 'left' | 'right';

export interface ChatBubbleContentProps extends LUIGlobalProps {}

export interface ChatBubbleFooterProps extends LUIGlobalProps {}

export interface SidebarCompoundComponent {
  Content: React.FunctionComponent<ChatBubbleContentProps>;
  Footer: React.FunctionComponent<ChatBubbleFooterProps>;
}

export interface ChatBubbleProps extends LUIGlobalProps {
  direction?: Direction;
}
