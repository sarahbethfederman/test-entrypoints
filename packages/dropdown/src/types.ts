import { LUIGlobalProps } from '@lendi-ui/utils';
import { BreakpointValueMap, BreakpointValue } from '@lendi-ui/breakpoint';
import { ThemeMap } from '@lendi-ui/theme';

export interface OptionType {
  [key: string]: any;
}
export type OptionsType<OptionType> = ReadonlyArray<OptionType>;
export type ValueType<OptionType> = OptionType | OptionsType<OptionType> | null | undefined;

export type Size = 'xs' | 'sm' | 'md' | 'lg';
export type InputSize = BreakpointValue<Size> | BreakpointValueMap<Size>;

export type FocusEventHandler = (event: React.FocusEvent<HTMLElement>) => void;
export type KeyboardEventHandler = (event: React.KeyboardEvent<HTMLElement>) => void;
export type FormatOptionLabelContext = 'menu' | 'value';

export interface LUISelectProps extends LUIGlobalProps {
  className?: string;
  /* Formats option labels in the menu and control as React components so that consumer can show menu-items with before and after icon or whatever way he wants to
  use it - see examples - `select-before-icon-example` and `select-with-description-example` */
  formatOptionLabel?: (option: OptionType, { context }: { context: FormatOptionLabelContext }) => React.ReactNode;
  /* Hide the selected option from the menu */
  hideSelectedOptions?: boolean;
  /* The value of the search input */
  inputValue?: string;
  isAutoFocus?: boolean;
  /* Is the select value clearable */
  isClearable?: boolean;
  /* Remove the currently focused option when the user presses backspace */
  isClearableByBackspace?: boolean;
  isDisabled?: boolean;
  isError?: boolean;
  isFullWidth?: boolean;
  isInverse?: boolean;
  isMenuOpen?: boolean;
  isOptionsOverflow?: boolean;
  /* enable multi select*/
  isMultiple?: boolean;
  isSearchable?: boolean;
  onBlur?: FocusEventHandler;
  onChangeItem?: (value: ValueType<OptionType>) => void;
  onFocus?: FocusEventHandler;
  onKeyDown?: KeyboardEventHandler;
  onInputChange?: (val: string) => string | void;
  onMenuOpen?: () => void;
  onMenuClose?: () => void;
  name?: string;
  /* Array of options that populate the select menu  - it is required*/
  options: OptionsType<OptionType>;
  placeholder?: string;
  size?: Size;
  theme?: ThemeMap;
  /* The value of the select; reflected by the selected option */
  value?: ValueType<OptionType>;
}
