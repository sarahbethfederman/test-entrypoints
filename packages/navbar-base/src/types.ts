import { FunctionComponent, ReactNode } from 'react';
import { LUIGlobalProps } from '@lendi-ui/utils';

export interface NavbarBaseProps extends LUIGlobalProps {
  isTransparent?: boolean;
  className?: string;
  children?: ReactNode;
}
export interface NavbarCompoundComponent {
  Left: FunctionComponent;
  Right: FunctionComponent;
}
export interface NavbarBaseSession {
  isTransparent?: boolean;
}

export interface NavbarBaseLeftProps extends LUIGlobalProps {
  children?: ReactNode;
}

export interface NavbarBaseRightProps extends LUIGlobalProps {
  children?: ReactNode;
}
