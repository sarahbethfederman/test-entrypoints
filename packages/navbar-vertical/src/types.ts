import { FunctionComponent } from 'react';
import { LUIGlobalProps } from '@lendi-ui/utils';

export interface VerticalNavbarProps extends LUIGlobalProps {}

export interface StyledVerticalNavbarProps {
  expanded: boolean;
}

export interface VerticalNavState {
  isMouseOver: boolean;
  isExpanded: boolean;
  setIsExpanded: () => void;
}

export interface VerticalNavbarCompoundComponent {
  Header: FunctionComponent<VerticalNavHeaderProps>;
  Tab: FunctionComponent<VerticalNavTabProps>;
}

export interface VerticalNavHeaderProps extends LUIGlobalProps {}

export interface VerticalNavTabProps extends LUIGlobalProps {
  selected?: boolean;
  tooltip?: string;
  onClick?: () => void;
  maxHeight?: string;
  disabled?: boolean;
}
