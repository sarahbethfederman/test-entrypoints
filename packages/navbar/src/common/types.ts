export enum ApplicationStage {
  started_application = 'started_application',
  finished_funnel2 = 'finished_funnel2',
  finished_funnel3 = 'finished_funnel3',
  finalised_application = 'finalised_application',
  loan_selected = 'loan_selected',
}

export type ApplicationType = 'Refinance' | 'New Purchase';

export interface Applicant {
  uuid?: string;
  sfid?: string;
  type?: 'primary' | string;
  firstName?: string;
}
export interface Application {
  type?: ApplicationType;
  sfid?: string;
  number?: number;
  stage?: ApplicationStage;
  createdDate?: string;
  continueURL?: string;
  applicants?: Applicant[];
  isAuth?: boolean;
}
export interface Broker {
  fullName: string;
  photo: string;
  phone: string;
  title: string;
}

export interface ApplicationInSF {
  activeApplicationNumber?: number;
  activeApplicationSfid?: string;
  applicants?: ApplicantInSF[];
  applicationStage?: string;
  applicationCreatedDate?: string;
  continueApplicationLink?: string;
  transactionType?: string;
  isAuth?: boolean;
}

export interface ApplicantInSF {
  applicant_type?: string;
  first_name?: string;
  married_to_uuid?: any;
  sfid?: string;
  uuid?: string;
}
