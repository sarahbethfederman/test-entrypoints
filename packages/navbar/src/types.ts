export type ApplicationStage =
  | 'no_application'
  | 'started_application'
  | 'finished_funnel2'
  | 'finished_funnel3'
  | 'finalised_application'
  | 'loan_selected';

export type ApplicationType = 'Refinance' | 'New Purchase';

export interface Applicant {
  uuid?: string;
  sfid?: string;
  type?: 'primary' | string;
}
export interface Application {
  type?: ApplicationType;
  sfid?: string;
  number?: number;
  stage?: ApplicationStage;
  continueURL?: string;
  applicants?: Applicant[];
}

export interface Broker {
  fullName: string;
  photo: string;
  phone: string;
  title: string;
}
