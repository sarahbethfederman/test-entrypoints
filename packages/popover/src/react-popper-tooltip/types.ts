import { Placement, Modifiers } from 'popper.js';
import * as React from 'react';
import { PopperArrowProps, RefHandler } from 'react-popper';

export interface GetTriggerPropsArg {
  onTouchEnd?(event: React.SyntheticEvent): void;
  onClick?(event: React.SyntheticEvent): void;
  onContextMenu?(event: React.SyntheticEvent): void;
  onMouseEnter?(event: React.SyntheticEvent): void;
  onMouseLeave?(event: React.SyntheticEvent): void;
  onMouseMove?(event: React.SyntheticEvent): void;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}

export interface GetTooltipPropsArg {
  style?: React.CSSProperties;
  onMouseEnter?: (event: React.SyntheticEvent) => void;
  onMouseLeave?: (event: React.SyntheticEvent) => void;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}

export interface GetArrowPropsArg {
  style?: React.CSSProperties;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  [key: string]: any;
}

export interface ChildrenArg {
  triggerRef: RefHandler;
  getTriggerProps(arg?: GetTriggerPropsArg): GetTriggerPropsArg;
}

export interface TooltipArg {
  arrowRef: RefHandler;
  tooltipRef: RefHandler;
  placement: Placement;
  getArrowProps(arg?: GetArrowPropsArg): GetArrowPropsArg;
  getTooltipProps(arg?: GetTooltipPropsArg): GetTooltipPropsArg;
}

export interface TooltipTriggerProps {
  /**
   * Whether to close the tooltip when it's trigger is out of the boundary
   * @default true
   */
  closeOnOutOfBoundaries: boolean;
  /**
   * Whether tooltip is shown by default
   * @default false
   */
  defaultTooltipShown: boolean;
  /**
   * Delay in hiding the tooltip
   * @default 0
   */
  delayHide: number;
  /**
   * Delay in showing the tooltip
   * @default 0
   */
  delayShow: number;
  /**
   * Whether to make the tooltip spawn at cursor position
   * @default false
   */
  followCursor: boolean;
  /**
   * Function that can be used to obtain a tooltip element reference
   */
  getTooltipRef?: RefHandler;
  /**
   * Function that can be used to obtain a trigger element reference
   */
  getTriggerRef?: RefHandler;
  /**
   * Modifiers passed directly to the underlying popper.js instance
   * For more information, refer to Popper.jsâ€™ modifier docs:
   * @link https://popper.js.org/popper-documentation.html#modifiers
   */
  modifiers?: Modifiers;
  /**
   * Tooltip placement w.r.t. trigger
   *  @default right
   */
  placement: Placement;
  /**
   * Element to be used as portal container
   * @default document.body
   */
  portalContainer: HTMLElement;
  /**
   * Used to create controlled tooltip
   */
  tooltipShown?: boolean;
  /**
   * Event that triggers the tooltip
   * @default hover
   */
  trigger: 'none' | 'click' | 'right-click' | 'hover';
  /**
   * Whether to use React.createPortal for creating tooltip
   * @default true // for browser environments
   */
  usePortal: boolean;
  /**
   * Trigger
   */
  children(arg: ChildrenArg): React.ReactNode;
  /**
   * Called when the visibility of the tooltip changes
   * @default no-op
   */
  onVisibilityChange(tooltipShown: boolean): void;
  /**
   * Tooltip
   */
  tooltip(arg: TooltipArg): React.ReactNode;
}

export interface TooltipTriggerState {
  pageX?: number;
  pageY?: number;
  tooltipShown: boolean;
}

export interface TooltipProps {
  arrowProps: PopperArrowProps;
  closeOnOutOfBoundaries: boolean;
  innerRef: RefHandler;
  outOfBoundaries: boolean | null;
  placement: Placement;
  style: React.CSSProperties;
  trigger: 'none' | 'click' | 'right-click' | 'hover';
  clearScheduled(): void;
  hideTooltip(): void;
  tooltip(arg: TooltipArg): React.ReactNode;
  scheduleUpdate(): void;
}
