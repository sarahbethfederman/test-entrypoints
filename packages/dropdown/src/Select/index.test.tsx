import * as React from 'react';
import Theme from '@lendi-ui/theme';
import { Select } from '../index';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import '@testing-library/react/cleanup-after-each';
import { MOCK_DATA_SOURCE } from '../examples/data-source.mock';
import { LUISelectProps } from '../types';
import { mount } from 'enzyme';
const singleValueTestProps: LUISelectProps = {
  options: MOCK_DATA_SOURCE,
  placeholder: 'single select',
  isClearable: true,
};

const multipleValueTestProps: LUISelectProps = {
  options: MOCK_DATA_SOURCE,
  placeholder: 'multiple select',
  isMultiple: true,
};

const mockFocus = jest.fn();
const mockBlur = jest.fn();
const mockOnChangeItem = jest.fn();
const mockOnInputChange = jest.fn();

const eventValueTestProps: LUISelectProps = {
  options: MOCK_DATA_SOURCE,
  placeholder: 'event testing',
  onFocus: mockFocus,
  onBlur: mockBlur,
  onChangeItem: mockOnChangeItem,
  onInputChange: mockOnInputChange,
};

const LUIfiedSelect = (args) => {
  return (
    <Theme>
      <Select {...args.value} />
    </Theme>
  );
};

describe('Select - the fancy one', () => {
  describe('should render with LUI custumizable cross and dropdown indicator', () => {
    afterEach(cleanup);
    it('should render dropdown icon', () => {
      const { container, getByTestId } = render(<LUIfiedSelect value={singleValueTestProps} />);
      expect(getByTestId('dropDownIcon')).toBeInTheDocument();
    });
    it('should render clear icon', () => {
      const { container, getByTestId } = render(<LUIfiedSelect value={singleValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      fireEvent.keyDown(theInput, { key: 'Enter' });
      expect(getByTestId('clearIcon')).toBeInTheDocument();
    });
  });
  describe('render single value - LUIfied select', () => {
    afterEach(cleanup);
    it('should render with testing placeholder', () => {
      const { getByText } = render(<LUIfiedSelect value={singleValueTestProps} />);
      expect(getByText('single select')).toBeInTheDocument();
    });
    it('should render input', () => {
      const { container } = render(<LUIfiedSelect value={singleValueTestProps} />);
      const theInput = container.querySelector('input');
      expect(theInput).toBeInTheDocument();
    });
    it('should all the styles with require snapshot', () => {
      const { container } = render(<LUIfiedSelect value={singleValueTestProps} />);
      expect(container).toMatchSnapshot();
    });
    it('should render menu', () => {
      const { container } = render(<LUIfiedSelect value={singleValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      const menu = container.parentElement.parentElement.querySelector("div[class*='-menu']");
      expect(menu).toBeInTheDocument();
    });
    it('should render required no of data item on keydown event', () => {
      const { container } = render(<LUIfiedSelect value={singleValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      const options = container.parentElement.parentElement.querySelectorAll("div[class*='-option']");
      expect(options.length).toEqual(MOCK_DATA_SOURCE.length);
    });
    it('should render filtered menu item on entering on inputs', () => {
      const { container } = render(<LUIfiedSelect value={singleValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.change(theInput, { target: { value: 'bank' } });
      const options = container.parentElement.parentElement.querySelectorAll("div[class*='-option']");
      expect(options.length).toEqual(16);
    });
    it('should be able to render single value select upon selecting one item', () => {
      const { container } = render(<LUIfiedSelect value={singleValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.change(theInput, { target: { value: 'bank' } });
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      fireEvent.keyDown(theInput, { key: 'Enter' });
      const singleValue = container.parentElement.parentElement.querySelector("div[class*='-singleValue']");
      expect(singleValue.innerHTML).toEqual('Adelaide Bank');
    });

    /* textOverflow:'ellipsis' is in css level so the label text won't have any difference when toggling isOptionsOverflow
        need to compare css attribute in this case.
        This simulates react-select to trigger the custom css of option
    */
    it("should display each option's text content with ellipsis overflow (...) when isOptionsOverflow is false in single select", () => {
      const wrapper = mount(<LUIfiedSelect value={{ ...singleValueTestProps, isOptionsOverflow: false }} />);

      const theInput = wrapper.find('input');
      theInput.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });

      const optionCss = wrapper.find('Option').first().get(0).props.selectProps.styles.option(true, true);
      expect(optionCss).toMatchObject({ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden' });
    });
    it("should display each option's full text content when isOptionsOverflow is true in single select", () => {
      const wrapper = mount(<LUIfiedSelect value={{ ...singleValueTestProps, isOptionsOverflow: true }} />);

      const theInput = wrapper.find('input');
      theInput.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });

      const optionCss = wrapper.find('Option').first().get(0).props.selectProps.styles.option(true, true);
      expect(optionCss).not.toMatchObject({ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden' });
    });
  });

  describe('render multiple value - LUIfied select', () => {
    afterEach(cleanup);
    it('should render with testing placeholder', () => {
      const { getByText } = render(<LUIfiedSelect value={multipleValueTestProps} />);
      expect(getByText('multiple select')).toBeInTheDocument();
    });
    it('should all the styles with require snapshot', () => {
      const { container } = render(<LUIfiedSelect value={multipleValueTestProps} />);
      expect(container).toMatchSnapshot();
    });

    it('should be able to render mulitple value select upon selecting multiple items', () => {
      const { container, debug } = render(<LUIfiedSelect value={multipleValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.change(theInput, { target: { value: 'bank' } });
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      fireEvent.keyDown(theInput, { key: 'Enter' });
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      fireEvent.keyDown(theInput, { key: 'Enter' });
      const multiValue = container.parentElement.parentElement.querySelectorAll("div[class*='-multiValue']");
      expect(multiValue.length).toEqual(2);
    });

    /* textOverflow:'ellipsis' is in css level so the label text won't have any difference when toggling isOptionsOverflow
        need to compare css attribute in this case.
        This simulates react-select to trigger the custom css of option
    */
    it("should display each option's text content with ellipsis overflow (...) when isOptionsOverflow is false in mulitple select", () => {
      const wrapper = mount(<LUIfiedSelect value={{ ...multipleValueTestProps, isOptionsOverflow: false }} />);

      const theInput = wrapper.find('input');
      theInput.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });

      const optionCss = wrapper.find('Option').first().get(0).props.selectProps.styles.option(true, true);
      expect(optionCss).toMatchObject({ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden' });
    });
    it("should display each option's full text content when isOptionsOverflow is true in mulitple select", () => {
      const wrapper = mount(<LUIfiedSelect value={{ ...multipleValueTestProps, isOptionsOverflow: true }} />);

      const theInput = wrapper.find('input');
      theInput.simulate('keyDown', { key: 'ArrowDown', keyCode: 40, which: 40 });

      const optionCss = wrapper.find('Option').first().get(0).props.selectProps.styles.option(true, true);
      expect(optionCss).not.toMatchObject({ textOverflow: 'ellipsis', whiteSpace: 'nowrap', overflow: 'hidden' });
    });
  });

  describe('event testing - LUIfied select', () => {
    afterEach(cleanup);
    it('should render with testing placeholder', () => {
      const { getByText } = render(<LUIfiedSelect value={eventValueTestProps} />);
      expect(getByText('event testing')).toBeInTheDocument();
    });
    it('should all the styles with require snapshot', () => {
      const { container } = render(<LUIfiedSelect value={eventValueTestProps} />);
      expect(container).toMatchSnapshot();
    });

    it('should call focus', () => {
      const { container } = render(<LUIfiedSelect value={eventValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.focus(theInput);
      expect(mockFocus).toHaveBeenCalled();
    });
    it('should call blur', () => {
      const { container } = render(<LUIfiedSelect value={eventValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.blur(theInput);
      expect(mockBlur).toHaveBeenCalled();
    });
    it('should call onChnageItem', () => {
      const { container } = render(<LUIfiedSelect value={eventValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.keyDown(theInput, { key: 'ArrowDown', keyCode: 40, which: 40 });
      fireEvent.keyDown(theInput, { key: 'Enter' });
      expect(mockOnChangeItem).toHaveBeenCalled();
    });
    it('should call onChangeInput', () => {
      const { container, debug } = render(<LUIfiedSelect value={eventValueTestProps} />);
      const theInput = container.querySelector('input');
      fireEvent.change(theInput, { target: { value: 'bank' } });
      expect(mockOnInputChange).toHaveBeenCalled();
    });
  });
});
