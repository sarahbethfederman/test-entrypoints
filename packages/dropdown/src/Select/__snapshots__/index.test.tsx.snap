// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select - the fancy one event testing - LUIfied select should all the styles with require snapshot 1`] = `
.c0 {
  color: #474747;
}

.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
  display: inline-block;
}

.c2 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c2 .fillPath {
  fill: rgba(0,192,165,1);
}

.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  -webkit-transition: -webkit-transform 0.5s ease 0s;
  -webkit-transition: transform 0.5s ease 0s;
  transition: transform 0.5s ease 0s;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      <div
        class=" css-lp3mvm-container"
      >
        <div
          class=" css-3vppv-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-ep7en5-placeholder"
            >
              event testing
            </div>
            <div
              class="css-8c79i4-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-19-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              style="display: flex; padding: 0px 8px 4px 4px; transition: color 150ms; box-sizing: border-box; cursor: pointer;"
            >
              <div>
                <span
                  class="c2 c3"
                  color="primary.500"
                  data-testid="dropDownIcon"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                    />
                    <path
                      d="M0-.75h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`Select - the fancy one render multiple value - LUIfied select should all the styles with require snapshot 1`] = `
.c0 {
  color: #474747;
}

.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
  display: inline-block;
}

.c2 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c2 .fillPath {
  fill: rgba(0,192,165,1);
}

.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  -webkit-transition: -webkit-transform 0.5s ease 0s;
  -webkit-transition: transform 0.5s ease 0s;
  transition: transform 0.5s ease 0s;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      <div
        class=" css-lp3mvm-container"
      >
        <div
          class=" css-3vppv-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-ep7en5-placeholder"
            >
              multiple select
            </div>
            <div
              class="css-8c79i4-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-14-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              style="display: flex; padding: 0px 8px 4px 4px; transition: color 150ms; box-sizing: border-box; cursor: pointer;"
            >
              <div>
                <span
                  class="c2 c3"
                  color="primary.500"
                  data-testid="dropDownIcon"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                    />
                    <path
                      d="M0-.75h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;

exports[`Select - the fancy one render single value - LUIfied select should all the styles with require snapshot 1`] = `
.c0 {
  color: #474747;
}

.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
  display: inline-block;
}

.c2 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c2 .fillPath {
  fill: rgba(0,192,165,1);
}

.c1 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c3 {
  -webkit-transition: -webkit-transform 0.5s ease 0s;
  -webkit-transition: transform 0.5s ease 0s;
  transition: transform 0.5s ease 0s;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      <div
        class=" css-lp3mvm-container"
      >
        <div
          class=" css-3vppv-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-ep7en5-placeholder"
            >
              single select
            </div>
            <div
              class="css-8c79i4-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-6-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              style="display: flex; padding: 0px 8px 4px 4px; transition: color 150ms; box-sizing: border-box; cursor: pointer;"
            >
              <div>
                <span
                  class="c2 c3"
                  color="primary.500"
                  data-testid="dropDownIcon"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 24 24"
                    width="1em"
                  >
                    <path
                      d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                    />
                    <path
                      d="M0-.75h24v24H0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </span>
  </div>
</div>
`;
